src/utils.rs

```diff
@@ -1,64 +1,112 @@
-use crossterm::style::Attribute;
 use crossterm::{
     cursor::MoveTo,
+    style::Attribute,
     terminal::{Clear, ClearType},
 };
-use std::fmt::Write;
-use std::io::prelude::*;
-use std::io::Write as IOWrite;
-use std::sync::MutexGuard;
 
-const LINE_NUMBERS: bool = false;
+use std::{
+    fmt::Write as _,
+    io::{self, Write as _},
+};
+
+const LINE_NUMBERS: LineNumbers = LineNumbers::No;
 
 pub(crate) fn draw(lines: String, rows: usize, upper_mark: &mut usize) {
-    // Split the String on each \n
-    let lines: Vec<&str> = lines.split_terminator('\n').collect();
-    // Calculate the lower mark
-    let mut lower_mark = *upper_mark + rows - 1;
-
-    // Do some necessory checking
-    // Lower mark should not be more than the lenght of lines vector
-    if lower_mark >= lines.len() {
-        lower_mark = lines.len();
+    let mut stdout = io::stdout();
+    let mut out = stdout.lock();
+
+    // Clear the screen and place cursor at the very top left
+    // FIXME(poliorcetics): handle result.
+    let _ = write!(&mut out, "{}{}", Clear(ClearType::All), MoveTo(0, 0));
+
+    // FIXME(poliorcetics): handle result.
+    let _ = write_lines(&mut out, &lines, rows, upper_mark, LINE_NUMBERS);
+
+    // Display the prompt
+    // FIXME(poliorcetics): handle result.
+    let _ = write!(
+        &mut out,
+        "{}{}Press q or Ctrl+C to quit{}",
+        MoveTo(0, rows as u16),
+        Attribute::Reverse,
+        Attribute::Reset,
+    );
+
+    drop(out);
+    let _ = stdout.flush();
+}
+
+/// Writes the given lines to the given `out`put.
+///
+/// - `rows` is the number of lines the `out`put can display at once.
```

Comment:
```suggestion
// - `rows` is the number of lines the terminal can display at once.
```

Rows is the number of rows the terminal can display at once and not the output

Reply:
The `out` parameter for this function is not necessarily a terminal, for example it's a `String` in the tests (that's what makes the function testable).